sdks/go/test/integration/io/xlang/kafka/jar.go

```diff
@@ -13,13 +13,18 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Exclude OSes which do not support the timeout command or syscall.SIGTERM.
```

Comment:
`timeout` is not available by default on Mac too. Can we check whether `timeout` is installed before `exec.Command` and halt gracefully if not? Or implement the timeout feature programmatically?

Reply:
AFAIK there's no way to implement timeout programmatically. Its purpose is to shut down the container if the test binary doesn't get a chance to do so. Checking for the timeout command sounds good though, I'll see if there's an OS-independent way to do that.

Reply:
Done, added checks for the timeout command (and the java command while I'm at it) and even added a way for people to run the test without timeout if their system doesn't support it, and added slightly better non-unix OS support.

As an aside, the ideal solution to this problem of Kafka cluster jars sticking around is to change our implementation of these jars to include an auto-shutdown within the jar itself. That way it'll always clean itself up, regardless of Beam SDK or OS used. Maybe I should make a Jira for that.